### PR TITLE
Use OutputBuilders from iota.go in transaction test framework

### PR DESCRIPTION
**Pull request by jkrvivian**
iota.go PR: https://github.com/iotaledger/iota.go/pull/532

### CODE REVIEW

File: pkg/testsuite/transactions_framework.go [hunk 1]
```diff
@@ -342,159 +329,107 @@ func (t *TransactionFramework) DefaultAddress() iotago.Address {
...
 }
 
-func WithBlockIssuerKeys(keys iotago.BlockIssuerKeys) options.Option[iotago.AccountOutput] {
-	return func(accountOutput *iotago.AccountOutput) {
-		blockIssuer := accountOutput.FeatureSet().BlockIssuer()
-		if blockIssuer == nil {
-			panic("cannot set block issuer keys to account without BlockIssuer feature")
-		}
-		blockIssuer.BlockIssuerKeys = keys
+func AddBlockIssuerKey(key iotago.BlockIssuerKey) options.Option[builder.AccountOutputBuilder] {
```
Comment (daria305) [inline pkg/testsuite/transactions_framework.go:376]: Shouldn't we use name  WithBlockIssuerKey to keep the convention? Then you need to update WithBlockIssuerKeys. Maybe WithSetBlockIssuerKeys, as both methods differ, becuase socond replace the keys, not add many

File: pkg/testsuite/transactions_framework.go [hunk 2]
```diff
@@ -342,159 +329,107 @@ func (t *TransactionFramework) DefaultAddress() iotago.Address {
...
+func WithAccountConditions(conditions iotago.AccountOutputUnlockConditions) options.Option[builder.AccountOutputBuilder] {
+	return func(accountBuilder *builder.AccountOutputBuilder) {
+		for _, condition := range conditions.MustSet() {
+			switch condition.Type() {
+			case iotago.UnlockConditionStateControllerAddress:
+				accountBuilder.StateController(condition.(*iotago.StateControllerAddressUnlockCondition).Address)
+			case iotago.UnlockConditionGovernorAddress:
+				accountBuilder.Governor(condition.(*iotago.GovernorAddressUnlockCondition).Address)
+			}
+		}
```
Comment (daria305) [inline pkg/testsuite/transactions_framework.go:426]: just checking if WithAccountIncreasedFoundryCounter, WithAccountIncreasedFoundryCounter and WithAccountImmutableFeatures were skipped on purpose?

**Review by daria305 — CHANGES_REQUESTED** [review]

File: pkg/testsuite/transactions_framework.go (hunk 1 above)
Comment (jkrvivian) [reply pkg/testsuite/transactions_framework.go:376]: Maybe the other way around, only this is adding a new element to the original value while others are replacing it. So I'll keep `WithBlockIssuerKeys` and rename `AddBlockIssuerKey` to `WithAddBlockIssuerKey`.

File: pkg/testsuite/transactions_framework.go (hunk 2 above)
Comment (jkrvivian) [reply pkg/testsuite/transactions_framework.go:426]: I removed it since it's not used during refactoring. :stuck_out_tongue: 
Will add it back.

**Review by daria305 — APPROVED** [review]

**Review by alexsporn — APPROVED** [review]

**Pull request merged**